torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -1247,6 +1248,31 @@ SegmentedGroup* SegmentCandidateFinder::mergeAllGivenGroups(
 
 namespace {
 
+inline void copyValue(
+    Val* key,
+    ExpressionEvaluator& from,
+    ExpressionEvaluator& to) {
+  auto concrete_val = from.evaluate(key);
+  TORCH_INTERNAL_ASSERT(concrete_val.has_value());
+  to.bind(key, concrete_val.value());
+}
+
+inline void inferGroupInputs(
+    SegmentedGroup* sg,
+    ExpressionEvaluator& ee,
+    ExpressionEvaluator& local_ee) {
+  for (auto v : getAllInputs(sg)) {
+    if (auto tv = dynamic_cast<TensorView*>(v)) {
+      for (auto id : tv->getRootDomain()) {
+        auto extent = id->extent();
```

Comment:
@naoyam with the shift PR can we still assume allocations based on extent or do we need to start using another interface?

Reply:
Allocation size is `id->extent()` + halo (if any). For example, https://github.com/csarofeen/pytorch/blob/shift-v1/torch/csrc/jit/codegen/cuda/lower_allocation.cpp#L150-L155